File: app/src/main/java/de/xikolo/storages/UserStorage.java
package de.xikolo.storages;

import android.content.Context;
import android.content.SharedPreferences;

import de.xikolo.storages.base.BaseStorage;

public class UserStorage extends BaseStorage {

    private static final String PREF_USER = "pref_user";

    private static String USER_ID = "id";
    private static String ACCESS_TOKEN = "token";

    public UserStorage() {
        super(PREF_USER, Context.MODE_PRIVATE);
    }

    public String getUserId() {
        return preferences.getString(USER_ID, null);
    }

    public void saveUserId(String id) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(USER_ID, id);
        editor.commit();
    }

    public String getAccessToken() {
        return preferences.getString(ACCESS_TOKEN, null);
    }

    public void saveAccessToken(String token) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(ACCESS_TOKEN, token);
        editor.commit();
    }

    public void delete() {
        SharedPreferences.Editor editor = preferences.edit();
        editor.clear();
        editor.commit();
    }

}

File: app/src/main/java/de/xikolo/storages/UserStorage.kt
package de.xikolo.storages

import android.content.Context
import de.xikolo.storages.base.BaseStorage

class UserStorage : BaseStorage(PREF_USER, Context.MODE_PRIVATE) {

    var userId: String?
        get() = getString(USER_ID)
        set(value) = putString(USER_ID, value)

    var accessToken: String?
        get() = getString(ACCESS_TOKEN)
        set(value) = putString(ACCESS_TOKEN, value)

    companion object {
        private const val PREF_USER = "pref_user_v2"
        private const val USER_ID = "id"
        private const val ACCESS_TOKEN = "token"
    }

}
